listings/applications/components/src/ownable.cairo

```diff
@@ -93,3 +96,146 @@ pub mod ownable_component {
+    use starknet::testing::{set_caller_address, set_contract_address};
+    use super::Errors;
+
+    fn deploy() -> (IOwnableDispatcher, ContractAddress) {
+        let (contract_address, _) = deploy_syscall(
+            MockContract::TEST_CLASS_HASH.try_into().unwrap(), 0, array![].span(), false
+        )
+            .unwrap_syscall();
+
+        (IOwnableDispatcher { contract_address }, contract_address)
+    }
+
+    #[test]
+    fn test_initial_state() {
+        let owner = contract_address_const::<'owner'>();
+        set_contract_address(owner);
+        let (ownable, _) = deploy();
+
+        assert(ownable.owner() == owner, 'wrong_owner');
+    }
+
+    #[test]
+    fn test_transfer_ownership_by_owner() {
+        set_contract_address(contract_address_const::<'initial'>());
+        let (ownable, _) = deploy();
+        let owner = contract_address_const::<'owner'>();
+        ownable.transfer_ownership(owner);
+
+        assert(ownable.owner() == owner, 'wrong_owner');
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_transfer_ownership_not_owner() {
+        set_contract_address(contract_address_const::<'initial'>());
+        let (ownable, _) = deploy();
+
+        set_contract_address(contract_address_const::<'not_owner'>());
+        ownable.transfer_ownership(contract_address_const::<'new_owner'>());
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_transfer_ownership_account_zero() {
+        set_contract_address(contract_address_const::<'initial'>());
+        let (ownable, _) = deploy();
+
+        ownable.transfer_ownership(Zero::zero());
+    }
+
+    #[test]
+    fn test_renounce_ownership() {
+        set_contract_address(contract_address_const::<'owner'>());
+        let (ownable, _) = deploy();
+
+        ownable.renounce_ownership();
+        assert(ownable.owner() == Zero::zero(), 'not_zero_owner');
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_renounce_ownership_previous_owner() {
+        set_contract_address(contract_address_const::<'owner'>());
+        let (ownable, _) = deploy();
+
+        ownable.renounce_ownership();
+        ownable.transfer_ownership(contract_address_const::<'new_owner'>());
+    }
+
+    #[test]
+    #[should_panic]
+    fn test_renounce_ownership_not_owner() {
+        set_contract_address(contract_address_const::<'owner'>());
+        let (ownable, _) = deploy();
+
+        set_contract_address(contract_address_const::<'not_owner'>());
+        ownable.renounce_ownership();
+    }
+
+    #[test]
+    fn test_ownership_renounced_event(){
+        let contract_address = contract_address_const::<'owner'>();
+        set_contract_address(contract_address);
+        let (ownable, address) = deploy();
+        ownable.renounce_ownership();
+        assert_eq!(
+            starknet::testing::pop_log(address),
+            Option::Some(Event::OwnershipRenouncedEvent(OwnershipRenouncedEvent { previous:contract_address }))
```

Comment:
```suggestion
            Option::Some(MockContract::Event::OwnableEvent(OwnershipRenouncedEvent { previous:contract_address }.into()))
```
Explainer: the issue is that you're expecting the component event type rather than the contract event type. Remember, `MockContract` emits the `Event` type of `MockContract`, whose first variant is itself an enum, specifically the `Event` enum of the component. The test now passes.

Reply:
This test can also be merged with `test_renounce_ownership`, since they are testing the same flow.